src/libraries/Native/Unix/System.Globalization.Native/pal_locale.c

```diff
@@ -46,11 +46,7 @@ int32_t GetLocale(const UChar* localeName,
     {
         UChar c = localeName[i];
 
-        // Some versions of ICU have a bug where '/' in name can cause infinite loop, so we preemptively
-        // detect this case for CultureNotFoundException (as '/' is anyway illegal in locale name and we
-        // expected ICU to return this error).
-
-        if (c > (UChar)0x7F || c == (UChar)'/')
+        if (c > (UChar)0x7F)
```

Comment:
>if [](start = 8, length = 2)

do you think we still need this condition? if we guarantee GetLocale is always called with locale name that already checked by InitIcuCultureDataCore, we can be ok converting this check into just assert.

Reply:
GetLocale has multiple call-sites in C code and some of them have fallback values sent from managed side. It was unclear whether removing the check is safe (there is also a branch below based on `UBool canonicalize`, which makes this function call a different ICU API when true).

Reply:
please keep the "/" check too. I saw some code paths can get into here directly from the user input. here is some example https://github.com/dotnet/runtime/blob/533a807b74c3b9258bb5355eb94c7fb48918f4b9/src/libraries/System.Private.CoreLib/src/System/Globalization/CultureInfo.cs#L1173

Reply:
The other idea could be inside the GetCultureInfo method, we call the validation. I looked at all GetLocale callsites and all of them at the end got from the already created culture which the name should be already validated. If we do that, we should convert this check to assert. I am ok not do this suggestion for now to not make the change complicated.

Reply:
I thought about it too. In fact, my first prototype of BCP47 was in C. After seeing the flow of how init does some preliminary validation, then some transformations on the input before the P/Invoke; I decided to keep it in managed  code. The rules to validate `const UChar* localeName` will be different as we can have `@` sign etc. when those transformations are applied.

Reply:
Let's keep what you are currently have. just add the `/` check in the native side.